lib/api/contracts/model_contract.rb

```diff
@@ -51,6 +51,27 @@ def self.attribute(*attributes, &block)
       validate :readonly_attributes_unchanged
       validate :run_attribute_validations
 
+      # Performs validation of both, the contract and the underlying model and returns
+      # the merged validation errors for both of them.
+      def validate_contract_and_model
+        self.validate
+        model.valid?
+
+        errors = self.errors.dup
```

Comment:
@NobodysNightmare We are in an instance, right? Wouldn't that mean that you take `errors` of the instance, dup them and then write the result back to `errors` of the instance. That would make it a NoOp. Haven't checked whether there is a setter for errors but even if what I say is wrong, you should employ a different name to avoid confusion.